src/main/java/io/github/battlepass/quests/workers/pipeline/steps/QuestValidationStep.java

```diff
@@ -115,11 +116,11 @@ public boolean isQuestPrimarilyValid(User user, Quest quest, BigInteger progress
         return !this.requirePreviousCompletion || isDaily || user.bypassesLockedWeeks() || !this.isPreviousWeekBlocked(user, week);
     }
 
-    public boolean isQuestValid(Player player, User user, Quest quest, BigInteger progress, boolean overrideUpdate) {
-        if (!this.isQuestPrimarilyValid(user, quest, progress, overrideUpdate)) {
+    public boolean isQuestValid(QuestExecution questExecution, Quest quest) {
+        if (!this.isQuestPrimarilyValid(questExecution.getUser(), quest, questExecution.getProgress(), questExecution.shouldOverrideUpdate())) {
             return false;
         }
-        String playerWorld = player.getWorld().getName();
+        String playerWorld = questExecution.getPlayer().getWorld().getName();
```

Comment:
Maybe pull "questExecution.getPlayer()" out to a local variable for clarity, nothing wrong with this though. And same with the instances of this above.

Reply:
Created variables for Player for both in the QuestValidationStep